### PR TITLE
Feature/create page my profile experience

### PR DESCRIPTION
**Pull request by LittleSweetGun**
Route.ts is not working for now (can't upload the form)

CREATE TABLE goodhive.users_experience (
    id SERIAL PRIMARY KEY,
    title VARCHAR(100) NOT NULL,
    type_employment VARCHAR(100),
    designation VARCHAR(100) NOT NULL,
    address VARCHAR(100) NOT NULL,
    contract_start VARCHAR(100) NOT NULL,
    contract_end VARCHAR(100) NOT NULL,
    description VARCHAR(255),
    skills VARCHAR(255)
);

### CODE REVIEW

File: app/components/select-input.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import React, { FC, useState, useRef } from "react";
+import { HiChevronDown } from "react-icons/hi";
```
Comment (n1c01a5) [inline app/components/select-input.tsx:2]: Don't use a tierces library for this. It's overengineering.

File: app/components/select-input.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+          <HiChevronDown className="text-gray-400" />
+        </div>
+      </div>
+      {isOptionsOpen && (
+        <ul className="absolute w-full z-10 mt-2 bg-white rounded-md shadow-md max-h-48 overflow-y-auto">
+          {filteredOptions.map((option) => (
+            <li
+              key={option.value}
+              className="py-2 px-4  cursor-pointer hover:bg-gray-100"
+              onClick={() => handleSelect(option)}
```
Comment (n1c01a5) [inline app/components/select-input.tsx:96]: onClick={(e) => setInputValue(e.target.value)}

File: app/components/select-input.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  required,
+  disabled,
+  value,
+  onChange,
+  options,
+}) => {
+  const [inputValue, setInputValue] = useState("");
+  const [isOptionsOpen, setIsOptionsOpen] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+
```
Comment (n1c01a5) [inline app/components/select-input.tsx:32]:
```
useEffect(() => {
  if(//click somewhere)  setIsOptionsOpen(!isOptionsOpen),
 }, [window]
)
```

File: app/design/page.tsx [hunk 4]
```diff
@@ -293,6 +304,12 @@ export default function Design() {
             value="Disabled value"
             onChange={textHandleChange}
           ></Input>
```
Comment (n1c01a5) [inline app/design/page.tsx:306]: />

File: app/talents/my-profile/experience/page.tsx [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+import { Button } from "../../../components/button";
+import { SelectInput } from "../../../components/select-input";
+
+interface Option {
+  value: string;
+  label: string;
+}
+export default function CreateJob() {
+  const [isLoading, setIsLoading] = useState(false);
+  const [value, setValue] = useState("");
```
Comment (n1c01a5) [inline app/talents/my-profile/experience/page.tsx:19]: Add types.

File: app/talents/my-profile/experience/page.tsx [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+    setIsLoading(true);
+
+    const formData = new FormData(e.currentTarget);
+
+    const contractStart = `${formData.get("start-month")}/${formData.get(
+      "start-year"
+    )}`;
+    const contractEnd = `${formData.get("end-month")}/${formData.get(
+      "end-year"
+    )}`;
```
Comment (n1c01a5) [inline app/talents/my-profile/experience/page.tsx:33]: Use Date Object.

File: app/talents/my-profile/experience/page.tsx [hunk 7]
```diff
@@ -0,0 +1,358 @@
...
+    { value: "2014", label: "2014" },
+    { value: "2015", label: "2015" },
+    { value: "2016", label: "2016" },
+    { value: "2017", label: "2017" },
+    { value: "2018", label: "2018" },
+    { value: "2019", label: "2019" },
+    { value: "2020", label: "2020" },
+    { value: "2021", label: "2021" },
+    { value: "2022", label: "2022" },
+    { value: "2023", label: "2023" },
```
Comment (n1c01a5) [inline app/talents/my-profile/experience/page.tsx:108]: Generate it with loop.

File: app/talents/my-profile/experience/page.tsx [hunk 8]
```diff
@@ -0,0 +1,358 @@
...
+      className:
+        "rounded-lg block w-full px-4 py-2 text-base font-normal text-gray-600 bg-clip-padding transition ease-in-out focus:text-black bg-gray-100 focus:outline-none focus:ring-0",
+    };
+
+    return (
+      <div className="relative">
+        <Autosuggest
+          suggestions={getSuggestions(value)}
+          onSuggestionsFetchRequested={onSuggestionsFetchRequested}
+          onSuggestionsClearRequested={onSuggestionsClearRequested}
```
Comment (n1c01a5) [inline app/talents/my-profile/experience/page.tsx:169]: setSuggestions([]);

File: package.json [hunk 9]
```diff
@@ -30,6 +30,7 @@
     "react-autosuggest": "^10.1.0",
     "react-dom": "18.2.0",
     "react-hot-toast": "^2.4.0",
+    "react-icons": "^4.8.0",
```
Comment (n1c01a5) [inline package.json:33]: Don't import this

**Review by n1c01a5 — CHANGES_REQUESTED** [review]

File: app/components/select-input.tsx [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+    const option = options.find((option) => option.value === selectedValue);
+    setInputValue(option || null);
+  };
+
+  const handleInputClick = (option: Option) => {
+    setInputValue(option.label);
+    setIsOptionsOpen(false);
+  };
+
+  console.log(inputValue);
```
Comment (n1c01a5) [inline app/components/select-input.tsx:46]: Remove this log

File: app/components/select-input.tsx [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+  placeholder: string;
+  required: boolean;
+  name?: string;
+  inputValue: string | null;
+  setInputValue: any;
+  onChange?: (value: string | null) => void;
+  disabled?: boolean;
+  options: Option[];
+}
+
```
Comment (n1c01a5) [inline app/components/select-input.tsx:19]: Add comment to explain this component it's not working as expected

File: app/constants/employmentType.js [hunk 12]
```diff
@@ -0,0 +1,7 @@
+export const employmentType = [
```
Comment (n1c01a5) [inline app/constants/employmentType.js:1]: Use `snake-case` for the name file.

File: app/talents/my-profile/experiences/page.tsx [hunk 13]
```diff
@@ -0,0 +1,350 @@
...
+import { employmentType } from "../../../constants/employmentType";
+
+interface Option {
+  value: string;
+  label: string;
+}
+export default function CreateJob() {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [selectedSkills, setSelectedSkills] = useState<string[]>([]);
+  //const [selectedOption, setSelectedOption] = useState<string | null>(null);
```
Comment (n1c01a5) [inline app/talents/my-profile/experiences/page.tsx:21]: Add comment to explain why this code is commented

File: app/talents/my-profile/experiences/page.tsx [hunk 14]
```diff
@@ -0,0 +1,350 @@
...
+    } else {
+      toast.success("Profile Saved!");
+    }
+  };
+
+  // Get the current date
+  const currentDate = new Date();
+
+  // Create an array of month options with full names
+  const month: Option[] = Array.from({ length: 12 }, (_, index) => {
```
Comment (n1c01a5) [inline app/talents/my-profile/experiences/page.tsx:69]: Can be simplify.

File: app/talents/my-profile/experiences/page.tsx [hunk 15]
```diff
@@ -0,0 +1,350 @@
...
+    const date = new Date(currentDate.getFullYear(), index);
+    const monthName = date.toLocaleString("en-US", { month: "long" });
+    return { value: monthName.toLowerCase(), label: monthName };
+  });
+
+  const startYear = parseInt(process.env.START_YEAR || "");
+  const endYear = parseInt(process.env.END_YEAR || "");
```
Comment (n1c01a5) [inline app/talents/my-profile/experiences/page.tsx:76]: Default value must be integer not string.

File: app/talents/my-profile/experiences/page.tsx [hunk 16]
```diff
@@ -0,0 +1,350 @@
...
+
+  const year = Array.from({ length: endYear - startYear + 1 }, (_, index) => {
```
Comment (n1c01a5) [inline app/talents/my-profile/experiences/page.tsx:78]: Can be simplify.

**Review by n1c01a5 — CHANGES_REQUESTED** [review]

File: app/talents/my-profile/experiences/page.tsx [hunk 17]
```diff
@@ -0,0 +1,350 @@
...
+      },
+      body: JSON.stringify(dataForm),
+    });
+
+    setIsLoading(false);
+
+    if (!experienceResponse.ok) {
+      toast.error("Something went wrong!");
+    } else {
+      toast.success("Profile Saved!");
```
Comment (n1c01a5) [inline app/talents/my-profile/experiences/page.tsx:61]: "Experience Saved!"

**Review by n1c01a5 — CHANGES_REQUESTED** [review]

File: package.json [hunk 18]
```diff
@@ -18,6 +18,7 @@
     "@typescript-eslint/eslint-plugin": "^5.58.0",
     "autoprefixer": "10.4.14",
     "aws-crt": "^1.15.16",
+    "buffer": "^6.0.3",
```
Comment (n1c01a5) [inline package.json:21]: Why this dep ?

**Review by n1c01a5 — CHANGES_REQUESTED** [review]

**Review by n1c01a5 — APPROVED** [review]

**Comment by LittleSweetGun** [comment]
#79 #74

**Pull request merged**